src/org/etools/j1939_84/controllers/StepController.java

```diff
@@ -153,6 +154,26 @@ protected void ensureKeyStateIs(KeyState requestedKeyState) throws InterruptedEx
         getListener().onResult("Final Engine Speed = " + getEngineSpeedAsString());
     }
 
+    protected void ensureKeyStateIs(KeyState requestedKeyState, String section) throws InterruptedException {
+        getListener().onResult("Initial Engine Speed = " + getEngineSpeedAsString());
+        if (getCurrentKeyState() != requestedKeyState) {
+            if (!isDevEnv()) {
+                getListener().onUrgentMessage("Please turn " + requestedKeyState,
+                                              "Step " + section,
+                                              WARNING,
+                                              getQuestionListener());
+            }
+            while (getCurrentKeyState() != requestedKeyState) {
+                updateProgress("Waiting for " + requestedKeyState + "...");
```

Comment:
This progress bar should include the section

Reply:
Re-review and let me know.